File: src/test/kotlin/katas/nakulgupta18/get_ascii_value_of_character.kt
package katas.nakulgupta18

import org.junit.Test
import kotlin.test.assertEquals

/**
 * get_ascii_value_of_character
 * @see <a href="https://www.codewars.com/kata/56a5d994ac971f1ac500003e/train/kotlin">nakulgupta18</a>
 *
 */

fun getAscii(c: Char): Int = c.code.toByte().toInt()


class TestGetAscii {
    @Test
    fun `Basic tests`() {
        assertEquals(expected = 65, getAscii('A'))
        assertEquals(expected = 32, getAscii(' '))
        assertEquals(expected = 33, getAscii('!'))
    }
}

File: src/test/javascript/katas/nakulgupta18/get_ascii_value_of_character.js
// Test.assertEquals(getASCII('A'),65)
// Test.assertEquals(getASCII(' '),32)
// Test.assertEquals(getASCII('!'),33)

const getASCII = c => c.toString().charCodeAt(0)

const c = 'A'
console.log(`getASCII(${c}): ${getASCII(c)}`)